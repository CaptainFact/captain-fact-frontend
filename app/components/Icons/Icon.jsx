import React, { Component } from 'react'
import PropTypes from 'prop-types';

const Icon = ({name}) => {
  return ( 
    <svg className={`i icon-${name}`}>
      <use xlinkHref={`#icon-${name}`}></use>
    </svg>
  )
}

Icon.propTypes = {
  name: PropTypes.string.isRequired
};

export default Icon;