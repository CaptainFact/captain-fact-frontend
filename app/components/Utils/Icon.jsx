import React from "react"

export class Icon extends React.PureComponent {
  constructor(props) {
    super(props)
  }

  renderIcon(name) {
    return (<i className={`fa icon-${name}`}/>)
  }

  render() {
    const {name, size, withContainer, className, isClickable, ...otherProps} = this.props
    const sizeClass = size ? ` is-${size} ` : ' '
    if (withContainer)
      if (isClickable) return (
        <a className={`icon${sizeClass}${className}`} {...otherProps}>
          {this.renderIcon(name)}
        </a>
      )
      else return (
        <span className={`icon${sizeClass}${className}`} {...otherProps}>
          {this.renderIcon(name)}
        </span>
      )
    else return this.renderIcon(name)
  }
}

Icon.defaultProps = {
  withContainer: true,
  isClickable: false,
  size: "",
  name: "question",
  className: ''
}
