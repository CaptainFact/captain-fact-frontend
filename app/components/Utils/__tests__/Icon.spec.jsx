import Icon from '../Icon'

test('render icon', () => {
  snapshot(<Icon name="plus"/>)
})

test('set size', () => {
  snapshot(<Icon name="plus" size="large"/>)
})

test('other props get passed to container', () => {
  snapshot(<Icon name="plus" title="Add some stuff"/>)
})
