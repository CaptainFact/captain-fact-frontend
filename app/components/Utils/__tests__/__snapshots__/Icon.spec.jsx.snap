// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`other props get passed to container 1`] = `
<span
  className="icon"
  title="Add some stuff"
>
  <RawIcon
    name="plus"
  />
</span>
`;

exports[`render icon 1`] = `
<span
  className="icon"
>
  <RawIcon
    name="plus"
  />
</span>
`;

exports[`set size 1`] = `
<span
  className="icon is-large"
>
  <RawIcon
    name="plus"
  />
</span>
`;
